storage/option.go

```diff
@@ -119,51 +158,33 @@ func (w *withDisabledClientMetrics) ApplyStorageOpt(c *storageConfig) {
 //
 // This is only supported for the read operation and that too for http(XML) client.
 // Grpc read-operation will be supported soon.
-//
-// Here, the input parameter decides the value of dynamic-timeout.
-// targetPercentile is the desired percentile of the observed latencies.
-// increaseRate determines the rate, timeout is adjusted. High means slow increase in
-// timeout and low means rapid increase.
-// initialTimeout decides the initial timeout.
-// minTimeout, maxTimeout is the lower & upper bound of the timeout.
-//
-// TODO (raj-prince): To keep separate dynamicDelay instance for different BucketHandle.
-// Currently, dynamicTimeout is kept at the client and hence shared across all the
-// BucketHandle, which is not not the ideal state. As latency depends on location of VM
-// and Bucket, and read latency of different buckets may lie in different range.
-// Hence hence having a separate dynamicTimeout instance at BucketHandle level will
-// be better
-func WithDynamicReadReqStallTimeout(
-	targetPercentile float64,
-	increaseRate float64,
-	initialTimeout time.Duration,
-	minTimeout time.Duration,
-	maxTimeout time.Duration) option.ClientOption {
+func WithDynamicReadReqStallTimeout(drrst *DynamicReadReqStallTimeoutConfig) option.ClientOption {
```

Comment:
Also, since we are using a struct now rather than args, we should choose sensible default values. I suggest 500 ms & 0.99.
> Yes, 500ms and 0.99 make sense. But this user has to set these value while using this option. Right?

Or do we want to expose the field as pointer so that we can set the right default if it's not set by user?

Reply:
I think we can leave it as values rather than pointers, and set to default values if zero values are detected for either parameter. Zero would not be a valid value for either param.